cmd/webhook-server/main.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"crypto/tls"
+	"fmt"
+	"github.com/coreos/pkg/capnslog"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	rookclient "github.com/rook/rook/pkg/client/clientset/versioned"
+	"github.com/rook/rook/pkg/clusterd"
+	"k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"net/http"
+	"path/filepath"
+)
+
+const (
+	tlsDir      = "/etc/webhook"
+	tlsCertFile = "tls.crt"
+	tlsKeyFile  = "tls.key"
+)
+
+var (
+	cephclusterResource = metav1.GroupVersionResource{Group: "ceph.rook.io", Version: "v1", Resource: "cephclusters"}
+	logger              = capnslog.NewPackageLogger("github.com/rook/rook", "webhookcmd")
+	rookClientSet       rookclient.Interface
+)
+
+func validateResource(req *v1beta1.AdmissionRequest) ([]patchOperation, error) {
```

Comment:
nit: `request` is preferred instead of `req` for readability

Reply:
Fixed